src/app/address/controllers/addressConfirm.js

```diff
@@ -37,10 +41,25 @@ class AddressConfirmController extends BaseController {
     });
   }
 
+  validateFields(req, res, callback) {
+    // add custom validator for houseName/Number check.
+    const formFields = req.form.options.fields;
+    const addresses = req.sessionModel.get("addresses");
+    const hasPreviousAddresses = addresses.length === 2 ? true : false;
+
+    formFields.isAddressMoreThanThreeMonths?.validate.push({
+      fn: confirmationValidation,
+      arguments: [hasPreviousAddresses],
+    });
+
+    super.validateFields(req, res, callback);
```

Comment:
Add this custom validator here due to the 'hasPreviousAddress' logic. 
As far as I'm aware there is no way to add this style argument to the fields argument.

Reply:
Yeah, I think this is another place where the direct reuse of the journey is causing us a few complications.

I think the more robust solution might be to extract out the page content themselves as components/templates, and then use separate "shell" steps to reuse parts as needed, with non-code config to link them up.